pkg/apis/certificates/validation/validation_test.go

```diff
@@ -0,0 +1,237 @@
+			},
+		},
+		"signerName path component is a symbol": {
+			csr: capi.CertificateSigningRequest{
+				ObjectMeta: validObjectMeta,
+				Spec: capi.CertificateSigningRequestSpec{
+					Usages:     validUsages,
+					Request:    newCSRPEM(t),
+					SignerName: "example.com/-",
+				},
+			},
+			errs: field.ErrorList{
+				field.Invalid(specPath.Child("signerName"), "example.com/-", "a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')"),
+			},
+		},
+		"signerName path component contains no '.' but is valid": {
+			csr: capi.CertificateSigningRequest{
+				ObjectMeta: validObjectMeta,
+				Spec: capi.CertificateSigningRequestSpec{
+					Usages:     validUsages,
+					Request:    newCSRPEM(t),
+					SignerName: "example.com/some-signer-name",
+				},
+			},
+			errs: field.ErrorList{},
+		},
+		"signerName with a length greater than 635 characters should be rejected": {
+			csr: capi.CertificateSigningRequest{
+				ObjectMeta: validObjectMeta,
+				Spec: capi.CertificateSigningRequestSpec{
+					Usages:  validUsages,
+					Request: newCSRPEM(t),
+					// this string is longer than the max signerName limit (635 chars)
+					SignerName: "thisisanexampleofareallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallylongsignername/it-is-653-characters-long",
+				},
+			},
+			errs: field.ErrorList{
+				field.TooLong(specPath.Child("signerName"), "thisisanexampleofareallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallyreallylongsignername/it-is-653-characters-long", 635),
+			},
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			el := ValidateCertificateSigningRequest(&test.csr)
```

Comment:
See earlier comment, but we def need some update tests as well.

Reply:
What should those update tests cover? We don't perform validation on Update as the registry already nils out changes

Reply:
I think we can ignore this per the defaulting on read thread.